SimulationAgent/Agent.cs

```diff
@@ -222,6 +227,7 @@ private void StopInactiveSimulations(IList<Simulation> activeSimulations)
 
             foreach (var manager in managersToStop)
             {
+                manager.Value.SaveStatisticsAsync();
```

Comment:
is `await` omitted on purpose? if so please add comment explaining the expected behavior when the tasks fail or timeout

Reply:
no, it was not omitted on purpose.. added await